x/concentrated-liquidity/pool_test.go

```diff
@@ -12,50 +12,182 @@ import (
-	expectedTokenIn := sdk.NewCoin("usdc", sdk.NewInt(0))
-	expectedTokenOut := sdk.NewCoin("eth", sdk.NewInt(42))
+	// limit max price impact to 5002 usdc per eth
+	priceLimit := sdk.NewDec(5002)
 
+	// run calculation
+	tokenInDelta, tokenOutDelta, updatedTick, updatedLiquidity, err := s.App.ConcentratedLiquidityKeeper.CalcOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, priceLimit, pool.Id)
 	s.Require().NoError(err)
-	s.Require().True(expectedTokenIn.Equal(tokenIn))
-	s.Require().True(expectedTokenOut.Equal(tokenOut))
-	// TODO: come back to tick calc: shouldn't this be 85183?
-	s.Require().Equal(sdk.NewInt(85184), updatedTick)
+
+	// we expect to put 42 usdc in and in return get .008398 eth back
+	expectedTokenInDelta := sdk.NewInt(42000000)
+	expectedTokenOutDelta := sdk.NewInt(8398)
+
+	// ensure tokenIn and tokenOut meet our expected values
+	s.Require().Equal(expectedTokenInDelta.String(), tokenInDelta.String())
+	s.Require().Equal(expectedTokenOutDelta.String(), tokenOutDelta.String())
+
+	// check the new tick is at the expected value
+	s.Require().Equal(sdk.NewInt(85180).String(), updatedTick.String())
+
+	// check pool liquidity
+	lowerSqrtPrice, err := s.App.ConcentratedLiquidityKeeper.TickToSqrtPrice(lowerTick)
+	s.Require().NoError(err)
+	upperSqrtPrice, err := s.App.ConcentratedLiquidityKeeper.TickToSqrtPrice(upperTick)
+	s.Require().NoError(err)
+	expectedLiquidity := cl.GetLiquidityFromAmounts(currSqrtPrice, lowerSqrtPrice, upperSqrtPrice, amount0.Mul(sdk.NewInt(2)), amount1.Mul(sdk.NewInt(2)))
+	s.Require().Equal(expectedLiquidity.String(), updatedLiquidity.String())
+
+	//
+	// TEST 2: one price range
+	//
+
+	// we use the same price range as above, but just with a single position instead of two
+
+	// 1 eth 5000 usdc position
+	amount0 = sdk.NewInt(1000000)
+	amount1 = sdk.NewInt(5000000000)
+
+	// create pool
+	pool, err = s.App.ConcentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(ctx, 2, "eth", "usdc", currSqrtPrice, currTick)
+	s.Require().NoError(err)
+
+	// add position
+	_, _, _, err = s.App.ConcentratedLiquidityKeeper.CreatePosition(ctx, pool.Id, s.TestAccs[1], amount0, amount1, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick.Int64(), upperTick.Int64())
+	s.Require().NoError(err)
+
+	// swapping parameters used for test
+	// swap in 42 usdc for some amount of eth
+	tokenIn = sdk.NewCoin("usdc", sdk.NewInt(42000000))
+	tokenOutDenom = "eth"
+	// set no swap fee
+	swapFee = sdk.ZeroDec()
+	// limit max price impact to 5004 usdc per eth
+	priceLimit = sdk.NewDec(5004)
+
+	// run calculation
+	tokenInDelta, tokenOutDelta, updatedTick, updatedLiquidity, err = s.App.ConcentratedLiquidityKeeper.CalcOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, priceLimit, pool.Id)
+	s.Require().NoError(err)
+
+	// we expect to put 41999999 usdc in and in return get .008396 eth back
+	expectedTokenInDelta = sdk.NewInt(41999999)
```

Comment:
This doesnt seem like the "delta" but the actual amount of token in?

Reply:
It is the delta. Token in delta is what the pool supply changes by, and the token out is also what the pool supply changes by (what you get out), but we take the abs value of that which is why they are both positive

Reply:
Changed the name